DeepSpeech.py

```diff
@@ -597,57 +573,38 @@ def get_git_branch():
 # Helpers
 # =======
 
-def calculate_and_print_wer_report(caption, results_tuple):
-    r"""
-    This routine will print a WER report with a given caption.
-    It'll print the `mean` WER plus summaries of the ``ds_report_count`` top lowest
-    loss items from the provided WER results tuple
-    (only items with WER!=0 and ordered by their WER).
-    """
-    items = list(zip(*results_tuple))
-
-    count = len(items)
+def calculate_report(results_tuple):
+    r'''
+    This routine will calculate a WER report.
+    It'll compute the `mean` WER and create ``Sample`` objects of the ``report_count`` top lowest
+    loss items from the provided WER results tuple (only items with WER!=0 and ordered by their WER).
+    '''
+    samples = []
+    items = zip(*results_tuple)
     mean_wer = 0.0
-    for i in range(count):
-        item = items[i]
-        # If distance > 0 we know that there is a WER > 0 and have to calculate it
-        if item[2] > 0:
-            # Replace result by language model corrected result
-            item = (item[0], correction(item[1]), item[2], item[3])
-            # Replacing accuracy tuple entry by the WER
-            item = (item[0], item[1], wer(item[0], item[1]), item[3])
-            # Replace items[i] with new item
-            items[i] = item
-            mean_wer = mean_wer + item[2]
+    for item in items:
+        sample = Sample(item[0], item[1], item[3], item[2])
+        samples.append(sample)
```

Comment:
This loop should be doing `mean_wer += sample.wer` to update the epoch mean WER stat. Without that we just report 0 WER for all epochs.